### ISSUE TITLE
Rant: KeepMinDuplicatePokemon

### ISSUE THREAD

**Issue by Majamulli** [NECROBOTIO/NecroBot#2612]
Alright I just had to write this here cause I am not quite happy with how this is being handled right now.
Until and including version 0.5.0 setting KeepMinDuplicatePokemon to 1 meant that you have 1 of each pokemon, we were used to this.
With 0.6.0 and the implementation of TransferWeakPokemon the KeepMinDuplicatePokemon was changed. If set to 1 it would mean that 2 of each would be kept. This confused many of us and a lot of people had full inventories. But okay, after a lot of talking here and in discord it was confirmed that this was indeed changed. So I set it to 0. And honestly, some information would have been nice that this was changed.
With the 0.6.4 update it was changed AGAIN and in the other direction. KeepMinDuplicatePokemon set to 1 means that it will keep 1 of each pokemon again, like it did before version 0.6.0. BUT - and this is really annoying me - many people, me included, already set it to 0! Starting the bot means it transfers all the pokemon you only have 1 each which don't meet the requirement. And that should never happen with a new version!

I am ranting here not only because of the changing back and forth and because it transfered some of my pokemon, I am ranting because I get the feeling that the developers don't have a common guideline to follow. I was told many times today that KeepMinDuplicatePokemon is right the way it worked in 0.6.0 - 0.6.3 because it's DUPLICATE pokemon. True that, can understand, but I can't understand that it gets changed back in the next update after it was defended by so many people.

I would be grateful if this was properly discussed between all developers and also properly told to the users which way it's going to stay. Changing it back and forth will result in unneccessary confusion and frustration and it might (like today) send the pokemon away that we want to keep!

Thanks and sorry for the rant.

**Comment by Fazy5** [comment]
I assumed that "keepminduplicatepokemon" to 0 would mean that no duplicates are kept and the original (only 1 pokemon) is kept as per the config requirement, the rest are transfered. and Im glad i was right.. Its working fine for me, since i didnt want "any" duplicates... Perfect!
setting it to 1 would mean that a minimum of "1" should be kept , and the maximum can be any number as long as the pokemon fulfill the criteria.

**Comment by Menton** [comment]
Well ... My bot just transfered my best Pikachu and some other pokemon because I didn't realise it instantly. I'm not running the bot anymore until I know that it works again. I recently posted a bug report about IV not being sorted correctly and losing strong pokemons off of it. I'm not really eager to lose anymore pokemons while figuring out what the problem could be.

**Comment by Bobguy123** [comment]
KeepMinDuplicatePokemon  =0  reads Keep a mininum of zero DUPLICATE of any 1 pokemon so this is very confusing and the change of it needing to be set to 1 again in 6.4 also caused me to lose pokemon as well. After update the bot needs to warn you of changes like this with a Y or N for do you want to continue imo.

**Comment by Fazy5** [comment]
My config works fine.. My settings are

"KeepMinCp": 5000,
  "KeepMinIvPercentage": 100.0,
  "KeepMinOperator": "or",
  "PrioritizeIvOverCp": false,
  "KeepMinDuplicatePokemon": 0,
"TransferWeakPokemon": false,
  "TransferDuplicatePokemon": true,
  "TransferDuplicatePokemonOnCapture": true,

**Comment by PokemasterMike** [comment]
the name is chosen poorly... see #2618 
close this and move the discussion to discord

**Comment by Bobguy123** [comment]
Have you updated to 6.4 fazy5? If not your in for a surpise.

**Comment by ytbro00** [comment]
On the previous updates we're supposed to put it as "KeepMinDuplicatePokemon = 1" If we want to keep at least 1 duplicate. On the last update they change this value to 0 so people had full inventory.. Now this new update we're supposed to have KeepMinDuplicatePokemon = 1. Like what it has always been before. Now people lost their only duplicates.. RIP

tl;dr.. KeepMinDuplicatePokemon =0 will now keep none. It's supposed to be KeepMinDuplicatePokemon =1 now like it has always been before.

**Comment by Fazy5** [comment]
Keeping no duplicates is all i want :/ 1 of each pokemon is enough ... I dont think how is that a surprise ?

**Comment by Majamulli** [comment]
@Fazy5 because with 0.6.4 your setting wouldn't keep 1 of each. It would transfer everything that doesnt meet 5000 CP or 100 IV, so it would transfer your Bulbasaur, your Ivysaur, your Venusaur and so on...

That's the reason why I made this thread. When you upgrade to 0.6.4 don't forget to set KeepMinDuplicatePokemon to 1 before starting the bot or you'll lose a lot of pokemon.

**Comment by Fazy5** [comment]
Oh Thanks a Ton @Majamulli  !

**Comment by alpadev** [comment]
Mh I can understand that some people are unhappy if they lose their pokemons but seriously you are using a bot (that is under heavy development by a bunch of different developers)..

If niantic will find a way to detect it and ban the hell out of us you'll also lose all of them maybe even losing the opportunity to use a bot also - who knows..

The devs here putting a lot effort in to this and you are just complaining that from time to time it's changing its behaviour and thus you're losing some of your precious botted pokemon - if you can do it better fork it and get your hands dirty on it..

jm2c

**Comment by Menton** [comment]
@alpadev you're right. The bot transfered some of our pokemon we only pocess thanks to him, so it shouldn't be bothered too much. However, I think this is far from "just complaining", since he actually made some sort of suggestion, even if it was accompanied by some saltiness. At least he admitted it.

Nevertheless, I like what the devs are capable of doing and I'd love to have some knowledge myself. I enjoy the fact that many more settings were added in order to improve the bot. Still, I agree with Majamulli that some sort of, let me call it improved changelog would be awesome.

**Comment by alpadev** [comment]
@Menton I'm totally with you..

I also lost almost all of my pokemons yesterday because of changes that got introduced into 0.6.0 but since this is a community project and not Necrobot LLC (that has some QA and so on) poeple should calm down in case something goes wrong and rethink if they should be really that mad about what happened..

Without that bot we would have to run around and stick to some broken tracking while wasting a lot of time trying to catch them all..

I know that there is still a lot room for improvements but ranting isn't helping at all..

**Comment by Menton** [comment]
@alpadev I agree with you in every point. What you're saying is actually one of the reasons I'm using this bot, it's because the game is not yet working proper enough to be time efficient and fun at the same time. Since I'm quite busy these days I really am gratefull to find a good bot I can use for free. Like 10 minutes after losing my pokemons I already calmed down and realised how silly it was of me to be upset.

**Comment by Ent19** [comment]
logfile (sry the names are in Ger, but look at the values!!)
[14:25:34] Anzahl gesehener Pokémon: 125/151, Anzahl gefundener Pokémon: 124/151False
[14:25:36] Hypno	- WP: 341  IV: 65.38%   [Beste WP: 1412  IV: 89.67%] (Bonbons: 1582)False
[14:25:41] Traumato	- WP: 715  IV: 86.62%   [Beste WP: 206  IV: 91.90%] (Bonbons: 1582)False
[14:25:42] Golbat	- WP: 301  IV: 70.72%   [Beste WP: 301  IV: 70.72%] (Bonbons: 1128)False
[14:25:43] Bluzuk	- WP: 316  IV: 89.60%   [Beste WP: 316  IV: 89.60%] (Bonbons: 384)False
[14:25:44] Owei	- WP: 611  IV: 87.08%   [Beste WP: 611  IV: 87.08%] (Bonbons: 37)False
[14:25:46] Nebulak	- WP: 557  IV: 88.39%   [Beste WP: 557  IV: 88.39%] (Bonbons: 148)False
[14:25:47] Rossana	- WP: 666  IV: 82.76%   [Beste WP: 682  IV: 96.02%] (Bonbons: 173)False
[14:25:49] Zubat	- WP: 238  IV: 31.80%   [Beste WP: 358  IV: 91.03%] (Bonbons: 1128)False
[14:25:50] NidoranM	- WP: 466  IV: 87.19%   [Beste WP: 466  IV: 87.19%] (Bonbons: 284)False
[14:25:50] Schiggy	- WP: 560  IV: 87.38%   [Beste WP: 560  IV: 87.38%] (Bonbons: 45)False
[14:25:51] Quapsel	- WP: 443  IV: 89.67%   [Beste WP: 441  IV: 87.92%] (Bonbons: 248)False
